### PR TITLE
heat solver notebooks update

### PR DESCRIPTION
**Pull request by dbochkov-flexcompute**
Updating heat notebooks to use where possible:
- new convenient perturbation models as perturbations to n and k
- 2d simulation sizes
- unstructured data in custom medium
- batch mode submission

also fixed a coupled of typos in thermally modulated ring resonator:
- one inconsequential in TiN heat capacity
- one very consequential in convective transfer coefficient value (10^6 difference), so you will see different results

### CODE REVIEW

**Review by tomflexcompute — APPROVED** [review]
Thanks @dbochkov-flexcompute for the updates!

**Review by e-g-melo — APPROVED** [review]
Thanks @dbochkov-flexcompute! 

The new classes to specify thermo-optic coefficients help a lot in GUI. Also, thanks for reducing `dl_interface` in ThermallyTunedRingResonator.

File: MachZehnderModulator.ipynb
Comment (marc-flex) [inline MachZehnderModulator.ipynb:1]: Minor comment
- Text between cells 9 and 10: "Specifically, we will sample the give..."
- This one I'm not sure (I'd need an English speaker) but I feel like the text before cell 26 should finish with "..as follows."
- Out of curiosity, does running in single precision (cell 29) produce different results? I tried to run but I had errors with conflicting packages

File: ThermallyTunedRingResonator.ipynb
Comment (marc-flex) [inline ThermallyTunedRingResonator.ipynb:1]: Minor comments here as well:
- Text before cell 1 there's a repeated "of": The bottom of the wafer..."

**Review by marc-flex — APPROVED** [review]
Thanks Daniil for updating these!

**Pull request merged**